lib/ex_aws/s3/impl.ex

```diff
@@ -234,6 +234,20 @@ defmodule ExAws.S3.Impl do
     request(client, :post, bucket, "/?delete", body: body_binary, headers: %{"content-md5" => content_md5})
   end
 
+  def delete_all_objects(client, bucket, objects, opts \\ []) do
+    do_delete_all(client, bucket, objects, opts, [])
+  end
+
+  defp do_delete_all(_client, _bucket, [], _opts, acc) do
+    {:ok, Enum.reverse(acc)}
+  end
+  defp do_delete_all(client, bucket, objects, opts, acc) do
+    {objects, rest} = Enum.split(objects, 1000)
+    with {:ok, result} <- delete_multiple_objects(client, bucket, objects, opts) do
+      do_delete_all(client, bucket, rest, opts, [result | acc])
+    end
```

Comment:
It depends on how we want to handle errors. The current return signature is either `{:ok, all_the_successful_stuff}` or `{:error, first_encountered_error_message}`. With Enum.chunk you could have interspersed errors and successes in the event of a network failure part way through that recovers or something. Alternatively we also do a take_while but the issue there is that it'll simply stop returning when an error is encountered, instead of returning the error.

Thoughts?